components/carousel/script.js

```diff
@@ -0,0 +1,368 @@
+     */
+    set navArrowStepSize(value) {
+        this._navArrowStepSize = value;
+    }
+
+    /**
+     * Return the currently active element on the carousel.
+     * The counting starts from the element which is positioned at the leftmost side.
+     */
+    get current() {
+        return this._current;
+    }
+
+    /**
+     * Set the currently active item.
+     * @param {number} value
+     */
+    set current(value) {
+        this._current = value;
+    }
+
+    /**
+     * Return the current page size.
+     */
+    get pageSize() {
+        return this._pageSize;
+    }
+
+    /**
+     * Set the current page size.
+     * @param {number} value
+     */
+    set pageSize(value) {
+        this._pageSize = value;
+        this.resize();
+        this.createPaginationControls();
+    }
+
+    /**
+     * Hide one of the navigation arrows.
+     * @param {number} direction - 1 for right arrow, -1 for left.
+     * @returns {undefined}
+     */
+    hideArrow(direction) {
+        if (direction === 1) return this.rightArrow.classList.add('guic-carousel-hidden');
+        this.leftArrow.classList.add('guic-carousel-hidden');
+    }
+
+    /**
+     * Show one of the navigation arrows.
+     * @param {number} direction - 1 for right arrow, -1 for left.
+     * @returns {undefined}
+     */
+    showArrow(direction) {
+        if (direction === 1) return this.rightArrow.classList.remove('guic-carousel-hidden');
+        this.leftArrow.classList.remove('guic-carousel-hidden');
+    }
+
+    /**
+     * Check if the next item will overflow the carousel's boundaries.
+     * @param {number} itemsLength - the maximum number of items in the carousel.
+     * @param {number} next - the index of the next element
+     * @returns {boolean}
+     */
+    overflows(itemsLength, next) {
+        return itemsLength - next < this.pageSize;
+    }
+
+    /**
+     * Get the index of the next element based on the current step size.
+     * @param {number} stepSize
+     * @returns {number}
+     */
+    nextStepIndex(stepSize) {
+        return (this.current + stepSize) % this.items.length;
+    }
+
+    /**
+     * Get the index of the previous element based on the current step size.
+     * @param {number} stepSize
+     * @returns {number}
+     */
+    prevStepIndex(stepSize) {
+        const itemsLength = this.items.length;
+        return (this.current - stepSize + itemsLength) % itemsLength;
+    }
+
+    /**
+     * Check if the up nav arrow should be visible - if navigating up is possible.
+     * If not - hide it.
+     * @returns {undefined}
+     */
+    shouldShowUpNavArrow() {
+        if (this.overflows(this.items.length, this.nextStepIndex(this.navArrowStepSize))) return this.hideArrow(1);
+        this.showArrow(1);
+    }
+
+    /**
+     * Check if the down nav arrow should be visible - if navigating down is possible.
+     * If not - hide it.
+     * @returns {undefined}
+     */
+    shouldShowDownNavArrow() {

```

Comment:
?
```
    shouldShowRightNavArrow() {
and
    shouldShowLeftNavArrow() {
```

Also, can these 2 functions be combined as one?

Reply:
Renamed to `shouldShowNavArrow` and combined to 1.